### PR TITLE
Fix/145 submission handle invalid

### PR DESCRIPTION
**Pull request by jaserud**
will PR after: https://github.com/icgc-argo/argo-clinical/pull/156
related PR in gateway: comming soon...

### CODE REVIEW

File: src/submission/submission-api.ts [hunk 1]
```diff
@@ -25,11 +25,22 @@ export enum FileType {
   SAMPLE = 'sample',
 }
 
-export const FileNameRegex = {
-  [FileType.REGISTRATION]: '^registration.*.tsv',
-  [FileType.DONOR]: '^(donor).*.tsv',
-  [FileType.SPECIMEN]: '^(specimen).*.tsv',
-  [FileType.SAMPLE]: '^(sample).*.tsv',
+const FileNameRegex = {
+  [FileType.REGISTRATION]: '^registration.*\\.tsv',
```
Comment (joneubank) [inline src/submission/submission-api.ts]: any reason we arent setting these as regex immediately?

`[FileType.REGISTRATION]:/^registration.*\.tsv/,`

File: src/submission/submission-api.ts [hunk 2]
```diff
@@ -109,10 +120,11 @@ class SubmissionController {
     if (!isValidCreateBody(req, res)) {
       return;
     }
-    const filesByTypeMap = mapFilesByType(req, res);
-    if (!filesByTypeMap) {
+    if (req.files == undefined || req.files.length == 0) {
```
Comment (joneubank) [inline src/submission/submission-api.ts]: ===

File: src/submission/submission-api.ts [hunk 3]
```diff
@@ -138,10 +151,12 @@ class SubmissionController {
       programId: req.params.programId,
     };
     const result = await submission.operations.uploadMultipleClinical(command);
-    if (result.successful) {
-      return res.status(200).send(result);
+    if (!result.successful) {
+      return res.status(422).send({ ...result, fileErrors: errorList });
+    } else if (errorList.length > 0) {
+      return res.status(207).send({ ...result, fileErrors: errorList });
```
Comment (joneubank) [inline src/submission/submission-api.ts]: Literally never seen 207 before. Very interesting.

Comment (jaserud) [reply src/submission/submission-api.ts]: :smile: honestly its just so we have some way to differentiate between the three cases

File: src/submission/submission-api.ts [hunk 4]
```diff
@@ -120,7 +120,7 @@ class SubmissionController {
     if (!isValidCreateBody(req, res)) {
       return;
     }
-    if (req.files == undefined || req.files.length == 0) {
+    if (req.files === undefined || req.files.length === 0) {
```
Comment (joneubank) [inline src/submission/submission-api.ts]: breathtaking

**Pull request merged**